src/bio/bio_context.c

```diff
@@ -694,14 +700,26 @@ bio_blob_unmap(struct bio_io_context *ioctxt, uint64_t off, uint64_t len)
 	rc = ba.bca_rc;
 	ioctxt->bic_inflight_dmas--;
 
-	if (rc != 0)
+	mea->mea_inflights = 1;
+	mem.mem_unmap = true;
+	mem.mem_bs = ioctxt->bic_xs_ctxt->bxc_blobstore;
+	mem.mem_tgt_id = ioctxt->bic_xs_ctxt->bxc_xs_id;
+
+	if (rc) {
+		spdk_thread_send_msg(owner_thread(mem.mem_bs), bio_media_error,
+				     &mem);
+		/* Wait for media error event to complete */
+		media_error_completion(ioctxt->bic_xs_ctxt, mea);
+
 		D_ERROR("Unmap blob %p failed for xs: %p rc:%d\n",
 			ioctxt->bic_blob, ioctxt->bic_xs_ctxt, rc);
-	else
+	} else
 		D_DEBUG(DB_MGMT, "Successfully unmapped blob %p for xs:%p\n",
 			ioctxt->bic_blob, ioctxt->bic_xs_ctxt);
 
+	media_error_cp_arg_fini(mea);
```

Comment:
Just need to send a media error on failure. (spdk_thread_send_msg(owner_thread, bio_media_error_event, ...)).

Reply:
I think this was only sending on error? I'll change to only alloc media_error_msg on error as well

Reply:
Yes, sending this message only on unmap error.